library/Zend/Form/Form.php

```diff
@@ -497,6 +497,12 @@ public function isValid()
             ));
         }
 
+        foreach ($filter->getInputs() as $input) {
+            if (method_exists($input, 'getName') && !$this->has($input->getName())) {
+                $filter->remove($input->getName());
```

Comment:
There is no need to remove elements from the input filter, you only have to change the validation group. The input filter itself can remain intact.